src/compiler/checker.ts

```diff
@@ -44,6 +44,7 @@ namespace ts {
 
         let compilerOptions = host.getCompilerOptions();
         let languageVersion = compilerOptions.target || ScriptTarget.ES3;
+        var modulekind = compilerOptions.module ? compilerOptions.module : languageVersion === ScriptTarget.ES6 ? ModuleKind.ES6 : ModuleKind.None;

```

Comment:
Perhaps it would be better to remove `ModuleKind.ES6`, as it could be confusing to API consumers, and instead use this here:
```ts
let moduleKind = compilerOptions.module || ModuleKind.None;
let nativeModules = languageVersion >= ScriptTarget.ES6 && moduleKind === ModuleKind.None;
```

Then use the `nativeModules` boolean variable when checking below, or merely inline the expression used to compute it.